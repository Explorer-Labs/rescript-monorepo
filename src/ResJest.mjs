// Generated by ReScript, PLEASE EDIT WITH CARE

import * as Curry from "rescript/lib/es6/curry.js";

function describe$1(string, $$function) {
  return describe(string, (function () {
                return Curry._1($$function, undefined);
              }));
}

function describeOnly(string, $$function) {
  return describe.only(string, (function () {
                return Curry._1($$function, undefined);
              }));
}

function describeSkip(string, $$function) {
  return describe.skip(string, (function () {
                return Curry._1($$function, undefined);
              }));
}

function test$1(string, $$function) {
  return test(string, (function () {
                return Curry._1($$function, undefined);
              }));
}

function testOnly(string, $$function) {
  return test.only(string, (function () {
                return Curry._1($$function, undefined);
              }));
}

function testSkip(string, $$function) {
  return test.skip(string, (function () {
                return Curry._1($$function, undefined);
              }));
}

function testTodo(string) {
  return test.todo(string);
}

function expect$1(rule) {
  return expect(rule);
}

function beforeEach$1($$function) {
  return beforeEach(function () {
              return Curry._1($$function, undefined);
            });
}

function afterEach$1($$function) {
  return afterEach(function () {
              return Curry._1($$function, undefined);
            });
}

var it = test$1;

var itOnly = testOnly;

var itSkip = testSkip;

export {
  describe$1 as describe,
  describeOnly ,
  describeSkip ,
  test$1 as test,
  testOnly ,
  testSkip ,
  testTodo ,
  it ,
  itOnly ,
  itSkip ,
  expect$1 as expect,
  beforeEach$1 as beforeEach,
  afterEach$1 as afterEach,
  
}
/* No side effect */
