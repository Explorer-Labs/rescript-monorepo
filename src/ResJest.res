let describe = (string, function) => Jest.describe(.string, (. ()) => function())
let describeOnly = (string, function) => Jest.describeOnly(.string, (. ()) => function())
let describeSkip = (string, function) => Jest.describeSkip(.string, (. ()) => function())

let test = (string, function) => Jest.test(.string, (. ()) => function())
let testOnly = (string, function) => Jest.testOnly(.string, (. ()) => function())
let testSkip = (string, function) => Jest.testSkip(.string, (. ()) => function())
let testTodo = string => Jest.testTodo(. string)

let it = test
let itOnly = testOnly
let itSkip = testSkip

let expect = rule => Jest.expect(. rule)

// @val external afterAll: (unit => unit) => unit = "afterAll"
// @val external afterEach: (. (. unit) => unit) => unit = "afterEach"
// @val external beforeAll: (. (. unit) => unit) => unit = "beforeAll"
// @val external beforeEach: (. (. unit) => unit) => unit = "beforeEach"

let beforeEach = function => Jest.beforeEach(.(. ()) => function())
let afterEach = function => Jest.afterEach(.(. ()) => function())

// external describeEach: (. array<array<'a>>, . string, (. 'a, 'a, 'a) => unit) => unit = "each"
// @scope("describe") @val external describeOnly: (. string, (. unit) => unit) => unit = "only"
// @scope(("describe", "only")) @val
// external describeOnlyEach: (. array<array<'a>>, . string, (. 'a, 'a, 'a) => unit) => unit = "each"
// @val external describeSkip: (. string, (. unit) => unit) => unit = "describe.skip"
