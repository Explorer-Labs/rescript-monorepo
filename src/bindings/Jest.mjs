// Generated by ReScript, PLEASE EDIT WITH CARE


var it = test;

var itOnly = test.only;

var itSkip = test.skip;

var itEach = test.each;

var itOnlyEach = test.only.each;

var fitEach = test.only.each;

function toHaveBeenCalled(prim0, prim1) {
  prim0.toBeCalled();
  
}

function notToHaveBeenCalled(prim0, prim1) {
  prim0.not.toBeCalled();
  
}

var Mock = {};

export {
  it ,
  itOnly ,
  itSkip ,
  itEach ,
  itOnlyEach ,
  fitEach ,
  toHaveBeenCalled ,
  notToHaveBeenCalled ,
  Mock ,
  
}
/* it Not a pure module */
