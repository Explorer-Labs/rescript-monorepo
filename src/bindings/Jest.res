type undefined = Js.undefined<unit>
type null = Js.null<unit>

type expectResult
type re = Js.Re.t

@val external describe: (. string, (. unit) => unit) => unit = "describe"
@val external test: (. string, (. unit) => unit) => unit = "test"
@val external it: (. string, (. unit) => unit) => unit = "it"
@val external expect: (. 'a) => expectResult = "expect"

@send external toBe: (expectResult, 'a) => unit = "toBe"
@scope("not") @send external notToBe: (expectResult, 'a) => unit = "toBe"

@send external toEqual: (expectResult, 'a) => unit = "toEqual"
@scope("not") @send external notToEqual: (expectResult, 'a) => unit = "toEqual"

@send external toContain: (expectResult, 'a) => unit = "toContain"
@scope("not") @send external notToContain: (expectResult, 'a) => unit = "toContain"

@send external toMatch: (expectResult, re) => unit = "toMatch"
@scope("not") @send external notToMatch: (expectResult, re) => unit = "toMatch"

@val external arrayContaining: array<'a> => expectResult = "expect.arrayContaining"
@val external notArrayContaining: array<'a> => expectResult = "expect.not.arrayContaining"

@val external objectContaining: Js.t<'a> => expectResult = "expect.objectContaining"
@val external notObjectContaining: Js.t<'a> => expectResult = "expect.not.objectContaining"

@val external stringContaining: string => expectResult = "expect.stringContaining"

@val external stringMatching: re => expectResult = "expect.stringMatching"

@send external toBeCalled: (expectResult, unit) => unit = "toBeCalled"
@scope("not") @send external notToBeCalled: (expectResult, unit) => unit = "toBeCalled"

// #region NOTE: Javascript Specific
@send external toBeNull: (expectResult, unit) => unit = "toBeNull"
@scope("not") @send external notToBeNull: (expectResult, unit) => unit = "toBeNull"

@send external toBeDefined: (expectResult, unit) => unit = "toBeDefined"
@scope("not") @send external notToBeDefined: (expectResult, unit) => unit = "toBeDefined"

@send external toBeUndefined: (expectResult, unit) => unit = "toBeUndefined"
@scope("not") @send external notToBeUndefined: (expectResult, unit) => unit = "toBeUndefined"

@send external toBeTruthy: (expectResult, unit) => unit = "toBeTruthy"
@scope("not") @send external notToBeTruthy: (expectResult, unit) => unit = "toBeTruthy"

@send external toBeFalsy: (expectResult, unit) => unit = "toBeFalsy"
@scope("not") @send external notToBeFalsy: (expectResult, unit) => unit = "toBeFalsy"
// #endregion

@send external toBeGreaterThan: (expectResult, 'a) => unit = "toBeGreaterThan"
@send external toBeGreaterThanOrEqual: (expectResult, 'a) => unit = "toBeGreaterThanOrEqual"
@send external toBeLessThan: (expectResult, 'a) => unit = "toBeLessThan"
@send external toBeLessThanOrEqual: (expectResult, 'a) => unit = "toBeLessThanOrEqual"
@send external toBeCloseTo: (expectResult, 'a) => unit = "toBeCloseTo"
