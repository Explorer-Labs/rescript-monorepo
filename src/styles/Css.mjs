// Generated by ReScript, PLEASE EDIT WITH CARE


function declaration(property, value) {
  return property + ": " + value + ";";
}

function toGlobalValues(globalValues) {
  if (globalValues === "unset") {
    return "unset";
  } else if (globalValues === "inherit") {
    return "inherit";
  } else {
    return "initial";
  }
}

function toBaselinePosition(baselinePosition) {
  if (baselinePosition === "firstBaseline") {
    return "first baseline";
  } else if (baselinePosition === "baseline") {
    return "baseline";
  } else {
    return "last baseline";
  }
}

function toContentDistribution(contentDistribution) {
  if (contentDistribution === "spaceAround") {
    return "space-around";
  } else if (contentDistribution === "stretch") {
    return "stretch";
  } else if (contentDistribution === "spaceBetween") {
    return "space-between";
  } else {
    return "space-evenly";
  }
}

function toContentPosition(contentPosition) {
  if (contentPosition === "start") {
    return "start";
  } else if (contentPosition === "center") {
    return "center";
  } else if (contentPosition === "flexStart") {
    return "flex-start";
  } else if (contentPosition === "flexEnd") {
    return "flex-end";
  } else {
    return "end";
  }
}

function toSelfPosition(selfPosition) {
  if (selfPosition === "selfEnd") {
    return "self-end";
  } else if (selfPosition === "start") {
    return "start";
  } else if (selfPosition === "center") {
    return "center";
  } else if (selfPosition === "flexStart") {
    return "flex-start";
  } else if (selfPosition === "flexEnd") {
    return "flex-end";
  } else if (selfPosition === "selfStart") {
    return "self-start";
  } else {
    return "end";
  }
}

function toValue(value) {
  if (value === "initial") {
    return "initial";
  } else if (value === "safe") {
    return "safe";
  } else if (value === "spaceAround") {
    return "space-around";
  } else if (value === "unsafe") {
    return "unsafe";
  } else if (value === "lastBaseline") {
    return "last baseline";
  } else if (value === "stretch") {
    return "stretch";
  } else if (value === "end") {
    return "end";
  } else if (value === "start") {
    return "start";
  } else if (value === "center") {
    return "center";
  } else if (value === "firstBaseline") {
    return "first baseline";
  } else if (value === "baseline") {
    return "baseline";
  } else if (value === "spaceBetween") {
    return "space-between";
  } else if (value === "unset") {
    return "unset";
  } else if (value === "flexStart") {
    return "flex-start";
  } else if (value === "inherit") {
    return "inherit";
  } else if (value === "normal") {
    return "normal";
  } else if (value === "flexEnd") {
    return "flex-end";
  } else {
    return "space-evenly";
  }
}

var AlignContent = {
  toValue: toValue
};

function toValue$1(value) {
  if (value === "safe") {
    return "safe";
  } else if (value === "unsafe") {
    return "unsafe";
  } else if (value === "lastBaseline") {
    return "last baseline";
  } else if (value === "stretch") {
    return "stretch";
  } else if (value === "end") {
    return "end";
  } else if (value === "selfEnd") {
    return "self-end";
  } else if (value === "start") {
    return "start";
  } else if (value === "center") {
    return "center";
  } else if (value === "firstBaseline") {
    return "first baseline";
  } else if (value === "baseline") {
    return "baseline";
  } else if (value === "unset") {
    return "unset";
  } else if (value === "flexStart") {
    return "flex-start";
  } else if (value === "inherit") {
    return "inherit";
  } else if (value === "normal") {
    return "normal";
  } else if (value === "flexEnd") {
    return "flex-end";
  } else if (value === "selfStart") {
    return "self-start";
  } else {
    return "initial";
  }
}

var AlignItems = {
  toValue: toValue$1
};

var AlignSelf = {};

var All = {};

function toAngle(angle) {
  if (angle === "turn") {
    return "turn";
  } else if (angle === "deg") {
    return "deg";
  } else if (angle === "rad") {
    return "rad";
  } else {
    return "grad";
  }
}

var Angle = {
  toAngle: toAngle
};

function toValue$2(value) {
  if (value === "ms") {
    return "ms";
  } else {
    return "s";
  }
}

var Time = {
  toValue: toValue$2
};

function toValue$3(value) {
  if (value === "alternateReverse") {
    return "alternate-reverse";
  } else if (value === "normal") {
    return "normal";
  } else if (value === "alternate") {
    return "alternate";
  } else {
    return "reverse";
  }
}

var SingleAnimationDirection = {
  toValue: toValue$3
};

function toValue$4(value) {
  if (value === "paused") {
    return "paused";
  } else {
    return "running";
  }
}

var AnimationPlayState = {
  toValue: toValue$4
};

function toValue$5(value) {
  if (value === "none") {
    return "none";
  } else if (value === "backwards") {
    return "backwards";
  } else if (value === "forwards") {
    return "forwards";
  } else {
    return "both";
  }
}

var SingleAnimationFillMode = {
  toValue: toValue$5
};

function toValue$6(value) {
  if (typeof value === "object") {
    return String(Math.abs(value.VAL));
  } else {
    return "infinite";
  }
}

var SingleAnimationIterationCount = {
  toValue: toValue$6
};

var AnimationTimingFunction = {};

function toValue$7(value) {
  if (typeof value === "object") {
    return value.VAL;
  } else {
    return "none";
  }
}

var AnimationName = {
  toValue: toValue$7
};

function alignContent(value) {
  return declaration("align-content", toValue(value));
}

function alignItems(value) {
  return declaration("align-items", toValue(value));
}

function animationDelay(time) {
  return declaration("animation-delay", toValue$2(time));
}

export {
  declaration ,
  toGlobalValues ,
  toBaselinePosition ,
  toContentDistribution ,
  toContentPosition ,
  toSelfPosition ,
  AlignContent ,
  AlignItems ,
  AlignSelf ,
  All ,
  Angle ,
  Time ,
  SingleAnimationDirection ,
  AnimationPlayState ,
  SingleAnimationFillMode ,
  SingleAnimationIterationCount ,
  AnimationTimingFunction ,
  AnimationName ,
  alignContent ,
  alignItems ,
  animationDelay ,
  
}
/* No side effect */
