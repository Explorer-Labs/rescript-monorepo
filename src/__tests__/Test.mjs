// Generated by ReScript, PLEASE EDIT WITH CARE

import * as Curry from "rescript/lib/es6/curry.js";
import * as React from "react";
import * as ReactTestRenderer from "react-test-renderer";
import * as Css$RescriptMonorepo from "../styles/Css.mjs";
import * as Sum$RescriptMonorepo from "../Sum.mjs";
import * as Jest$RescriptMonorepo from "../bindings/Jest.mjs";
import * as Utils$RescriptMonorepo from "../utils/Utils.mjs";
import * as ResJest$RescriptMonorepo from "../ResJest.mjs";

function drinkAll(callback, flavour) {
  if (flavour !== "octopus") {
    return flavour;
  } else {
    return Curry._1(callback, flavour);
  }
}

var globalValue1 = {
  contents: 0
};

ResJest$RescriptMonorepo.describe("Snapshots", (function (param) {
        var styles = Css$RescriptMonorepo.declaration("float", "left");
        var Test = function (Props) {
          var children = Props.children;
          return React.createElement("h1", {
                      className: styles
                    }, children);
        };
        ResJest$RescriptMonorepo.test("renders with correct styles", (function (param) {
                var tree = ReactTestRenderer.create(React.createElement(Test, {
                            children: Utils$RescriptMonorepo.str("hello world")
                          })).toJSON();
                ResJest$RescriptMonorepo.expect(tree).toMatchSnapshot();
                
              }));
        
      }));

ResJest$RescriptMonorepo.describe("Common Matchers", (function (param) {
        ResJest$RescriptMonorepo.test("toBe", (function (param) {
                ResJest$RescriptMonorepo.expect(Sum$RescriptMonorepo.sum(1, 2)).toBe(3);
                
              }));
        ResJest$RescriptMonorepo.test("two plus two is four", (function (param) {
                ResJest$RescriptMonorepo.expect(4).toBe(4);
                
              }));
        ResJest$RescriptMonorepo.test("two plus two is not five", (function (param) {
                ResJest$RescriptMonorepo.expect(4).not.toBe(5);
                
              }));
        ResJest$RescriptMonorepo.test("object assignment using rescript records", (function (param) {
                var dataUpdate = {
                  num: 2,
                  name: "test"
                };
                var dataMutable = {
                  name: "Baby",
                  age: 5
                };
                dataMutable.age = dataMutable.age + 1 | 0;
                ResJest$RescriptMonorepo.expect({
                        num: 1,
                        name: "test"
                      }).toEqual({
                      num: 1,
                      name: "test"
                    });
                ResJest$RescriptMonorepo.expect(dataUpdate).toEqual({
                      num: 2,
                      name: "test"
                    });
                ResJest$RescriptMonorepo.expect(dataMutable).toEqual({
                      name: "Baby",
                      age: 6
                    });
                
              }));
        ResJest$RescriptMonorepo.test("null", (function (param) {
                var $$null = null;
                ResJest$RescriptMonorepo.expect($$null).toBeDefined();
                ResJest$RescriptMonorepo.expect(undefined).not.toBeDefined();
                ResJest$RescriptMonorepo.expect($$null).toBeNull();
                ResJest$RescriptMonorepo.expect(undefined).not.toBeNull();
                ResJest$RescriptMonorepo.expect(undefined).toBeUndefined();
                ResJest$RescriptMonorepo.expect($$null).not.toBeUndefined();
                ResJest$RescriptMonorepo.expect(true).toBeTruthy();
                ResJest$RescriptMonorepo.expect($$null).not.toBeTruthy();
                ResJest$RescriptMonorepo.expect($$null).toBeFalsy();
                ResJest$RescriptMonorepo.expect(true).not.toBeFalsy();
                
              }));
        ResJest$RescriptMonorepo.test("zero", (function (param) {
                ResJest$RescriptMonorepo.expect(0).not.toBeNull();
                ResJest$RescriptMonorepo.expect(0).toBeDefined();
                ResJest$RescriptMonorepo.expect(0).not.toBeUndefined();
                ResJest$RescriptMonorepo.expect(0).not.toBeTruthy();
                ResJest$RescriptMonorepo.expect(0).toBeFalsy();
                
              }));
        ResJest$RescriptMonorepo.test("two plus two", (function (param) {
                ResJest$RescriptMonorepo.expect(4).toBeGreaterThan(3);
                ResJest$RescriptMonorepo.expect(4).toBeGreaterThanOrEqual(3.5);
                ResJest$RescriptMonorepo.expect(4).toBeLessThan(5);
                ResJest$RescriptMonorepo.expect(4).toBeLessThanOrEqual(4.5);
                ResJest$RescriptMonorepo.expect(4).toBe(4);
                ResJest$RescriptMonorepo.expect(4).toEqual(4);
                
              }));
        ResJest$RescriptMonorepo.test("adding floating point numbers", (function (param) {
                var value = 0.1 + 0.2;
                ResJest$RescriptMonorepo.expect(value).toBeCloseTo(0.3);
                
              }));
        ResJest$RescriptMonorepo.test("the shopping list has beer on it", (function (param) {
                var shoppingList = [
                  "diapers",
                  "kleenex",
                  "trash bags",
                  "paper towels",
                  "beer"
                ];
                ResJest$RescriptMonorepo.expect(shoppingList).toContain("beer");
                ResJest$RescriptMonorepo.expect(shoppingList).not.toContain("BEER");
                
              }));
        ResJest$RescriptMonorepo.test("there is no I in team", (function (param) {
                ResJest$RescriptMonorepo.expect("team").not.toMatch(/I/);
                
              }));
        ResJest$RescriptMonorepo.test("but there is a 'stop' in Christoph", (function (param) {
                ResJest$RescriptMonorepo.expect("Christoph").toMatch(/stop/);
                
              }));
        ResJest$RescriptMonorepo.test("id should match object", (function (param) {
                var obj = {
                  id: "111",
                  product: "Test",
                  url: "https://jestjs.io/"
                };
                ResJest$RescriptMonorepo.expect(obj).toEqual(expect.objectContaining({
                          id: "111"
                        }));
                ResJest$RescriptMonorepo.expect(obj).toEqual(expect.not.objectContaining({
                          title: "Test"
                        }));
                
              }));
        ResJest$RescriptMonorepo.test("String containing", (function (param) {
                ResJest$RescriptMonorepo.expect("abcdefghijk").toEqual(expect.stringContaining("abcde"));
                ResJest$RescriptMonorepo.expect("abcdefghijk").not.toEqual(expect.stringContaining("abce"));
                
              }));
        ResJest$RescriptMonorepo.test("String matching", (function (param) {
                ResJest$RescriptMonorepo.expect("abcdefghijk").toEqual(expect.stringContaining("abcde"));
                ResJest$RescriptMonorepo.expect("abcdefghijk").not.toEqual(expect.stringContaining("abce"));
                
              }));
        
      }));

ResJest$RescriptMonorepo.describe("Array Containing", (function (param) {
        var expected = [
          "Alice",
          "Bob"
        ];
        ResJest$RescriptMonorepo.it("matches even if received contains additional elements", (function (param) {
                ResJest$RescriptMonorepo.expect([
                        "Alice",
                        "Bob",
                        "Eve"
                      ]).toEqual(expect.arrayContaining(expected));
                
              }));
        ResJest$RescriptMonorepo.it("matches even if received contains additional elements", (function (param) {
                ResJest$RescriptMonorepo.expect([
                        "Bob",
                        "Eve"
                      ]).toEqual(expect.not.arrayContaining(expected));
                
              }));
        ResJest$RescriptMonorepo.it("does not match if received does not contain expected elements", (function (param) {
                ResJest$RescriptMonorepo.expect([
                        "Bob",
                        "Eve"
                      ]).not.toEqual(expect.arrayContaining(expected));
                
              }));
        
      }));

ResJest$RescriptMonorepo.describe("stringMatching in arrayContaining", (function (param) {
        var expected = [
          expect.stringMatching(/^Alic/),
          expect.stringMatching(/^[BR]ob/)
        ];
        ResJest$RescriptMonorepo.it("matches even if received contains additional elements", (function (param) {
                ResJest$RescriptMonorepo.expect([
                        "Alicia",
                        "Roberto",
                        "Evelina"
                      ]).toEqual(expect.arrayContaining(expected));
                
              }));
        ResJest$RescriptMonorepo.it("does not match if received does not contain expected elements", (function (param) {
                ResJest$RescriptMonorepo.expect([
                        "Roberto",
                        "Evelina"
                      ]).not.toEqual(expect.arrayContaining(expected));
                
              }));
        
      }));

ResJest$RescriptMonorepo.describe("Mock function", (function (param) {
        ResJest$RescriptMonorepo.test("drinks something lemon-flavoured", (function (param) {
                var mockFn = jest.fn();
                Curry._1(mockFn, undefined);
                return Jest$RescriptMonorepo.toHaveBeenCalled(ResJest$RescriptMonorepo.expect(mockFn), undefined);
              }));
        
      }));

ResJest$RescriptMonorepo.describe("Mock function with implementation", (function (param) {
        ResJest$RescriptMonorepo.test("drinks something lemon-flavoured", (function (param) {
                var returnsTrue = jest.fn(function (param) {
                      return true;
                    });
                ResJest$RescriptMonorepo.expect(Curry._1(returnsTrue, undefined)).toBe(true);
                
              }));
        
      }));

ResJest$RescriptMonorepo.describe("Lifecycle", (function (param) {
        var persistentValue = {
          contents: 0
        };
        ResJest$RescriptMonorepo.beforeEach(function (param) {
              persistentValue.contents = persistentValue.contents + 1 | 0;
              
            });
        ResJest$RescriptMonorepo.afterEach(function (param) {
              persistentValue.contents = persistentValue.contents - 1 | 0;
              
            });
        ResJest$RescriptMonorepo.test("runs lifecycle the first time", (function (param) {
                ResJest$RescriptMonorepo.expect(persistentValue.contents).toBe(1);
                
              }));
        ResJest$RescriptMonorepo.test("runs lifecycle each time", (function (param) {
                ResJest$RescriptMonorepo.expect(persistentValue.contents).toBe(1);
                
              }));
        
      }));

ResJest$RescriptMonorepo.describe("Skips other tests", (function (param) {
        var persistentValue = {
          contents: 0
        };
        ResJest$RescriptMonorepo.beforeEach(function (param) {
              persistentValue.contents = persistentValue.contents + 1 | 0;
              
            });
        ResJest$RescriptMonorepo.afterEach(function (param) {
              persistentValue.contents = persistentValue.contents - 1 | 0;
              
            });
        ResJest$RescriptMonorepo.testSkip("runs lifecycle the first time", (function (param) {
                ResJest$RescriptMonorepo.expect(persistentValue.contents).toBe(1);
                
              }));
        ResJest$RescriptMonorepo.test("runs lifecycle each time", (function (param) {
                ResJest$RescriptMonorepo.expect(persistentValue.contents).toBe(1);
                
              }));
        
      }));

ResJest$RescriptMonorepo.describeSkip("Lifecycle", (function (param) {
        var persistentValue = {
          contents: 0
        };
        beforeEach(function () {
              persistentValue.contents = persistentValue.contents + 1 | 0;
              
            });
        afterEach(function () {
              persistentValue.contents = persistentValue.contents - 1 | 0;
              
            });
        test("runs lifecycle the first time", (function () {
                expect(persistentValue.contents).toBe(1);
                
              }));
        test("runs lifecycle each time", (function () {
                expect(persistentValue.contents).toBe(1);
                
              }));
        
      }));

describe("Test each functionality", (function () {
        test.each([
                [
                  1,
                  1,
                  2
                ],
                [
                  1,
                  2,
                  3
                ],
                [
                  2,
                  1,
                  3
                ]
              ])(".add(%i, %i)", (function (a, b, expected) {
                expect(a + b | 0).toBe(expected);
                
              }));
        Jest$RescriptMonorepo.itEach([
                [
                  1,
                  1,
                  2
                ],
                [
                  1,
                  2,
                  3
                ],
                [
                  2,
                  1,
                  3
                ]
              ])(".add(%i, %i)", (function (a, b, expected) {
                expect(a + b | 0).toBe(expected);
                
              }));
        
      }));

describe("Test skip each", (function () {
        test.skip.each([
                [
                  1,
                  1,
                  2
                ],
                [
                  1,
                  2,
                  3
                ],
                [
                  2,
                  1,
                  3
                ]
              ])(".add(%i, %i)", (function (a, b, expected) {
                expect(a + b | 0).toBe(expected);
                
              }));
        
      }));

describe("Todo works", (function () {
        test.todo("Write test 1");
        test.todo("Write test 2");
        test.todo("Write test 3");
        
      }));

describe.each([
        [
          1,
          1,
          2
        ],
        [
          1,
          2,
          3
        ],
        [
          2,
          1,
          3
        ]
      ])(".add(%i, %i)", (function (a, b, expected) {
        test("returns " + String(expected), (function () {
                expect(a + b | 0).toBe(expected);
                
              }));
        test("returned value not be greater than " + String(expected), (function () {
                expect(a + b | 0).not.toBeGreaterThan(expected);
                
              }));
        test("returned value not be less than " + String(expected), (function () {
                expect(a + b | 0).not.toBeLessThan(expected);
                
              }));
        
      }));

var sum = Sum$RescriptMonorepo.sum;

export {
  sum ,
  drinkAll ,
  globalValue1 ,
  
}
/*  Not a pure module */
