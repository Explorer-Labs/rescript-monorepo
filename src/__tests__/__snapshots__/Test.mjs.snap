// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots renders with correct styles 1`] = `
<h1>
  hello world!
</h1>
`;
